v3/src/components/graph/hooks/graph-hooks.ts

```diff
@@ -41,34 +43,42 @@ export const useGetData = (props: IUseGetDataProps) => {
     yScale = layout.axisScale("left")
 
   const result = useMemo(() => {
-    let xAttrId= '', yAttrId = '', xName = '', yName = ''
+    let xAttrId = '', yAttrId = '', xName = '', yName = ''
     const data: InternalizedData = {
       xAttributeID: '',
       yAttributeID: '',
       cases: []
     }
 
     const findNumericAttrIds = (attrsToSearch: IAttribute[]) => {
-      for (const iAttr of attrsToSearch) {
-        if (iAttr.type === 'numeric') {
-          if (xAttrId === '') {
-            xAttrId = iAttr.id
-          } else if (yAttrId === '') {
-            yAttrId = iAttr.id
-          } else {
-            break
+        for (const iAttr of attrsToSearch) {
+          if (iAttr.type === 'numeric') {
+            if (xAttrId === '') {
+              xAttrId = iAttr.id
+            } else if (yAttrId === '') {
+              yAttrId = iAttr.id
+            } else {
+              break
+            }
           }
         }
+      },
+
+      setNiceDomain = (values: number[], scale: ScaleBaseType, axis: INumericAxisModel) => {
+        const valueExtent = extent(values, d => d) as [number, number],
+          niceBounds = computeNiceNumericBounds(valueExtent[0], valueExtent[1])
+        scale.domain([niceBounds.min, niceBounds.max])
+        const [min, max] = scale.domain()
+        axis.setDomain(min, max)
```

Comment:
I see that you're following the pattern of the preexisting code, but it should be sufficient to set the domain on the axis model and let the synchronization to the scale happen via the normal update mechanism which would simplify the code and prove that the system is working as intended.

Reply:
Yes, that's true!